### ISSUE TITLE
feature: Allow specifying Docker build target

### ISSUE THREAD

**Issue by mikkelam** [aws/aws-sam-cli#2578]
### Description:
I'm trying to use the DockerBuildArgs option to use a specific target in my Dockerfile like so :

```
...
    Metadata:
      Dockerfile: Dockerfile
      DockerContext: .
      DockerBuildArgs:
        target: prod
...
```
However it does not work. I'm getting multiple warnings from docker that build args were not consumed

### Steps to reproduce:
Dockerfile like so
```
FROM public.ecr.aws/lambda/python:3.8 as prod

COPY app.py requirements.txt ./

RUN python3.8 -m pip install -r requirements.txt -t .

# Command can be overwritten by providing a different command in the template directly.
CMD ["app.lambda_handler"]

FROM prod as dev
CMD []
```
along with `sam build`

### Observed result:
```
`[Warning] One or more build-args [target] were not consumed`
```

### Expected result:
I would be able to use a specific target with sam

### Additional environment details (Ex: Windows, Mac, Amazon Linux etc)

1. OS: macos Big Sur
2. `sam --version`: SAM CLI, version 1.15.0

**Comment by mgrandis** [comment]
It seems the `DockerBuildArgs` param is used to pass parameters to the DockerFile at build time.
That's why you get the warning for unused/not consumed parameter.
We are not supporting specifying a `target` for now but we'll review adding this feature.
This would basically mean allowing a `DockerBuildTarget` element under `Metadata`.

Docker python client reference: https://docker-py.readthedocs.io/en/stable/images.html

**Comment by mgrandis** [comment]
Released in [1.20.0](https://github.com/aws/aws-sam-cli/releases/tag/v1.20.0)
